src/proof/dot/dot_printer.h

```diff
@@ -87,10 +145,47 @@ class DotPrinter
 
   /** Traverse proof node and populate d_lbind
    *
-   * @param pn the proof node to be traversed
+   * @param pn The proof node to be traversed
    */
   void letifyResults(const ProofNode* pn);
 
+  /** Define the proof node type and populate d_nodesClusterType and
+   * d_scopesArgs.
+   * @param pn The proof node to categorized.
+   * @return the current node type
+   */
+  NodeClusterType defineNodeType(const ProofNode* pn, NodeClusterType last);
+
+  /** Verify if the proof node is an input node
```

Comment:
Clarify the detail about the shadowing of scopes that requires the more intricate check